### DISCUSSION TITLE
ISR and On demande revalidation

### DISCUSSION THREAD

**Discussion by nabildroid** [vercel/next.js#51502] [Help] [answered]
### Summary

i'm wokring with Next 13.4.6 but never could achieve ISR with OnDemand revalidation,
not sure if it's because a bug in Next or i didn't implement it in right way

can you help me please achieve ISR
here is my repo https://github.com/nabildroid/tempocach
when you visit /site/hello it get cached
but wen calling the API /api/reva then revisit multiple time /site/hello, it never get revalidated

### Example

here is my repo https://github.com/nabildroid/tempocach

**Comment by PsyGik** [comment]
# ISR
- To use ISR, add the `revalidate` prop to `getStaticProps`

# On-Demand Revalidate

Inside `getStaticProps`, you do not need to specify `revalidate` to use on-demand revalidation. If `revalidate` is omitted, Next.js will use the default value of false (no revalidation) and only revalidate the page on-demand when `revalidate()` is called.

When running locally with `next dev`, `getStaticProps` is invoked on every request. To verify your on-demand ISR configuration is correct, you will need to create a production build and start the production server:

```sh
$ next build
$ next start
```

All the above from: https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration

**Reply by nabildroid** [reply]
Yes, in page router it works, but when using appRouter it doesn't work

**Reply by PsyGik** [reply]
Please share a [Codesandbox](https://codesandbox.io) with minimum reproducible code.

**Reply by nabildroid** [reply]
https://codesandbox.io/p/sandbox/next-js-fxis37

**Reply by nabildroid** [reply]
in dev it works fine, but in production the catch never get revalidated

**Comment by nabildroid** [comment, marked as answer]
solution 
https://github.com/vercel/next.js/discussions/51612#discussioncomment-6250425
